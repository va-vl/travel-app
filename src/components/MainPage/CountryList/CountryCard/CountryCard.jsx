import * as React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import CountryCardTextContainer from './CountryCardText/CountryCardTextContainer';

const CountryCard = ({
  id, name, capital, image,
}) => (
  <Link
    to={`/countryId=${id}`}
    className="card"
    style={{
      backgroundImage: `url('${image}')`,
    }}
  >
    <CountryCardTextContainer name={name} capital={capital} />
  </Link>
);

CountryCard.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  capital: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default CountryCard;
