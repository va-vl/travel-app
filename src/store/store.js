import { combineReducers, applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import countryListReducer from './countryListReducer/countryListReducer';
import countrySearchReducer from './countrySearchReducer/countrySearchReducer';

const store = createStore(combineReducers({
  countryListReducer,
  countrySearchReducer,
}), composeWithDevTools(applyMiddleware(thunk)));

export default store;
