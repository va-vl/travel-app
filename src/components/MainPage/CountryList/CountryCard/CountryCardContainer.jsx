import * as React from 'react';
import { useSelector } from 'react-redux';
import CountryCard from './CountryCard';

const CountryListContainer = (props) => {
  const { isLoading, isReady } = useSelector((state) => state.countryListReducer);

  return <CountryCard isLoading={isLoading} isReady={isReady} {...props} />;
};

export default CountryListContainer;
