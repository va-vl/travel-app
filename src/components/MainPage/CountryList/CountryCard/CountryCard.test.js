import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import CountryCard from './CountryCard';
import store from '../../../../store/store';

test('component CountryCard renders', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <BrowserRouter>
        <CountryCard></CountryCard>
      </BrowserRouter>
    </Provider>
    , div);
});

test('render correctly CountryCard component', () => {
  const CountryCardComponent = TestRenderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <CountryCard /></BrowserRouter>
    </Provider>
  ).toJSON();
  expect(CountryCardComponent).toMatchSnapshot();
});
