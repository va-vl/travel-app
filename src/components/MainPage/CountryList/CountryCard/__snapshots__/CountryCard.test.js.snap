// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly CountryCard component 1`] = `
<a
  className="card"
  href="/undefined"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url('undefined')",
    }
  }
/>
`;
